flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraphTest.java

```diff
@@ -149,9 +159,27 @@ public void testSuccessfulSlotAssignmentTransitionsToExecuting() throws Exceptio
                     actualExecutionGraph ->
                             assertThat(actualExecutionGraph, sameInstance(executionGraph)));
 
-            executionGraphWithvertexParallelismFuture.complete(
+            final AtomicReference<Throwable> handledFailureRef = new AtomicReference<>();
+            context.setCreateExecutingStateFunction(
+                    () ->
+                            new AdaptiveSchedulerTest.DummyState() {
+
+                                @Override
+                                public void handleGlobalFailure(Throwable cause) {
+                                    handledFailureRef.set(cause);
```

Comment:
This seems overly complicated; isn't it enough to know that it was indeed called?